test.sh

```diff
@@ -29,31 +30,58 @@ esac
 
 set -e
 
-if [ "$VALIDATE" -eq "1" ]; then
-# Validate --no-default-features build
-echo "________Validate build________"
-time cargo check --no-default-features
-time cargo check --manifest-path util/io/Cargo.toml --no-default-features
-time cargo check --manifest-path util/io/Cargo.toml --features "mio"
-
-# Validate chainspecs
-echo "________Validate chainspecs________"
-time ./scripts/validate_chainspecs.sh
-fi
-
-# Running the C++ example
-echo "________Running the C++ example________"
-cd parity-clib-examples/cpp && \
-  mkdir -p build && \
-  cd build && \
-  cmake .. && \
-  make -j 8 && \
-  ./parity-example && \
-  cd .. && \
-  rm -rf build && \
-  cd ../..
-
-# Running tests
-echo "________Running Parity Full Test Suite________"
-git submodule update --init --recursive
-time cargo test  $OPTIONS --features "$FEATURES" --all $1 -- --test-threads 8
+
+
+case $CARGO_TARGET in
+  (x86_64-unknown-linux-gnu|x86_64-apple-darwin|x86_64-pc-windows-msvc|'')
+    # native builds
+    if [ "$VALIDATE" -eq "1" ]
+    then
+      echo "________Validate build________"
+      time cargo check --no-default-features
+      time cargo check --manifest-path util/io/Cargo.toml --no-default-features
+      time cargo check --manifest-path util/io/Cargo.toml --features "mio"
```

Comment:
why do they not require a $TARGET?

Reply:
They use the default target. I don't want to break the script for those who run it locally and probably don't have the CARGO_TARGET variable set because it's not required and specific to the CI.